sphinx/screens-detail/payments/payment-template/payment-template/src/main/java/chat/sphinx/payment_template/ui/PaymentTemplateFragment.kt

```diff
@@ -0,0 +1,206 @@
+    protected lateinit var imageLoader: ImageLoader<ImageView>
+
+    private val args: PaymentTemplateFragmentArgs by navArgs()
+    override val viewModel: PaymentTemplateViewModel by viewModels()
+    override val binding: FragmentPaymentTemplateBinding by viewBinding(
+        FragmentPaymentTemplateBinding::bind
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.apply {
+            includePaymentTemplateHeader.apply {
+                textViewDetailScreenHeaderName.text = getString(R.string.payment_template_header_name)
+
+                textViewDetailScreenClose.setOnClickListener {
+                    lifecycleScope.launch(viewModel.mainImmediate) {
+                        viewModel.navigator.closeDetailScreen()
+                    }
+                }
+                textViewDetailScreenHeaderNavBack.visible
+                textViewDetailScreenHeaderNavBack.setOnClickListener {
+                    lifecycleScope.launch(viewModel.mainImmediate) {
+                        viewModel.navigator.popBackStack()
+                    }
+                }
+            }
+
+            textViewAmount.text = args.argAmount.toString()
+            textViewMessage.text = args.argMessage
+
+            buttonConfirm.setOnClickListener {
+                viewModel.sendPayment()
+            }
+        }
+
+        setupPaymentTemplateRecycler()
+        setupFragmentLayout()
+    }
+
+    private var snapPosition = RecyclerView.NO_POSITION
+    private fun setupPaymentTemplateRecycler() {
+        binding.recyclerViewPaymentTemplate.apply {
+            val paymentTemplateAdapter = PaymentTemplateAdapter(
+                this,
+                imageLoader,
+                onStopSupervisor,
+                viewModel,
+            )
+
+            layoutManager = LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
+            adapter = paymentTemplateAdapter
+
+            val snapHelper: SnapHelper = LinearSnapHelper()
+            snapHelper.attachToRecyclerView(this)
+
+            addOnScrollListener(object : RecyclerView.OnScrollListener() {
+                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                    super.onScrolled(recyclerView, dx, dy)
+
+                    snapHelper.findSnapView(layoutManager)?.let { centerView ->
+                        layoutManager?.getPosition(centerView)?.let { position ->
+                            if (snapPosition != position) {
+                                //Subtracting padding item and no template item from position
+                                viewModel.selectTemplate(position - 2)
+                            }
+                        }
+                    }
+                }
+            })
+        }
+
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.loadTemplateImages()
+        }
```

Comment:
Improper use of onStopSupervisor scope. If it helps inhibit further misuse of this scope due to lack of understanding Fragment Lifecycle events (stop/start), we should rename to `observeViewStateSupervisor`.

Remove coroutine usage and lift to ViewModel.